cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -298,41 +345,86 @@ def delete_if_exists(self):
         if self.exists:
             self.delete()
 
-    def add_button_group(self):
-        button_name = fauxfactory.gen_alpha()
-        view = navigate_to(self, 'AddButtonGroup')
-        view.fill({'btn_group_text': 'group_text',
-                   'btn_group_hvr_text': button_name,
-                   'btn_image': self.button_icon_name})
+    def add_button_group(self, **kwargs):
+        button_name = kwargs.get("text", "gp_{}".format(fauxfactory.gen_alpha()))
+        hover_name = kwargs.get("hover", "hover_{}".format(fauxfactory.gen_alpha()))
+        image = kwargs.get("image", "broom")
+
+        view = navigate_to(self, "AddButtonGroup")
+        view.fill(
+            {
+                "btn_group_text": button_name,
+                "btn_group_hvr_text": hover_name,
+                "btn_image": image,
+            }
+        )
+
         view.add.click()
-        view = self.create_view(DetailsCatalogItemView)
-        assert view.is_displayed
+        view = self.create_view(DetailsCatalogItemView, wait="10s")
         view.flash.assert_no_error()
         return button_name
 
-    def add_button(self):
-        button_name = fauxfactory.gen_alpha()
+    def button_group_exits(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        try:
+            view.catalog_items.tree.fill(path)
+            return True
+        except Exception as e:
+            logger.exception(e)
+            return False
+
+    def delete_button_group(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        view.catalog_items.tree.fill(path)
+        view.configuration.item_select("Remove this Button Group", handle_alert=True)
+
+        # ToDo: BZ-1687289 To avoid landing on buttons for catalog item page adding workaround.
```

Comment:
Optional: From what I understand about the BZ, I think as a workaround, if there is at least one item left after deletion, you can simply assert the "Delete Successful" message, but if there are no items left after deletion, you can check the success by checking tree path.

Reply:
we don't check success in model page. Its part of test. best work around is assert no error in flash.